src/player.cpp

```diff
@@ -488,6 +488,9 @@ void player::process_turn()
     if ( (has_trait("FLOWERS")) && (!(has_trait("CHLOROMORPH"))) ) {
         norm_scent -= 200;
     }
+	//Handle scent effects.
+    norm_scent -= scent_mod;
```

Comment:
Here you subtract scent mod, but in ink, you have negative scent mod. It looks to me like it will result in ink increasing scent.